src/shared/api/services/axios.ts

```diff
@@ -7,4 +14,50 @@ const axiosService = axios.create({
 
 axiosService.defaults.headers.common["Content-Type"] = "application/json"
 
+axiosService.interceptors.request.use(
+  config => {
+    const tokens = secureTokensStorage.getTokens()
+
+    if (tokens?.accessToken && tokens?.tokenType) {
+      config.headers!.Authorization = `${tokens.tokenType} ${tokens.accessToken}`
+    }
+
+    return config
+  },
+  error => {
+    return Promise.reject(error)
+  },
+)
+
+axiosService.interceptors.response.use(
+  response => response,
+  async (error: AxiosError) => {
+    const config = error?.config as RequestConfig
+
+    if (error.response?.status === 401 && !config?.retry) {
+      config.retry = true
+
+      const tokens = secureTokensStorage.getTokens()
+
+      if (!tokens?.refreshToken || !tokens.tokenType) {
+        return Promise.reject(error)
+      }
+
+      try {
+        const { data } = await authorizationService.refreshToken({ refreshToken: tokens?.refreshToken })
+
+        secureTokensStorage.setTokens(data.result)
+
+        config.headers!.Authorization = `${data.result.tokenType} ${data.result.accessToken}`
+      } catch (e) {
+        Promise.reject(e)
```

Comment:
What happens if refresh token is expired?

Reply:
Done